### ISSUE TITLE
[TECH ED] APIs

### ISSUE THREAD

**Issue by areebsattar** [areebsattar/My-Coursework-Planner#209]
### Link to the coursework

https://github.com/CodeYourFuture/Module-Node

### Why are we doing this?

Think carefully and try to answer the following questions:

1. Define an interface? (Think about the answer in multiple contexts, including programming )
1. Explain why do we need to use an interface in programming? (Advanced)
1. List 3 things that are part of the NodeJS API 
1. List 3 things that are part of the Jest API
1. List 3 things that are part of the Spotify API
1. Explain why Jest is not a web-based API and why Spotify is a web-based API.

### Maximum time in hours

.5

### How to get help

⏰ Don't spend longer than 30 minutes on this exercise 
📝 Make a note of any questions you've struggled to answer. Raise these on Slack before Saturday's session so mentors know what to focus on

### How to submit

Post in the thread on Slack. If there is no thread yet, start one.

### How to review

Compare your answers during study group in class. What can you learn from each other?

### Anything else?

If you use ChatGPT to answer these questions, you have missed the point of this course and you are now fully wasting everyone's time. Don't use LLM for this assignment. The point of this assignment is to find out what you, the human person, need help with in class.

**Comment by areebsattar** [comment]
**Define an interface? (Think about the answer in multiple contexts, including programming)**
Ans. In Simpler terms interface is a connection between two computers or a computer and a human. But based on the context it can mean different things as well for example in programming its a data type which acts as an abstraction of class
**Explain why do we need to use an interface in programming? (Advanced)**
Ans. Interfaces usually allow the different objects to expose the common functionality and it can help us write shorter and simpler code to make it work.
**List 3 things that are part of the NodeJS API**
- Process: to retrieve information about environment variables, CPU usage and reporting
- Events: Work similarly to the client side event listeners
- File System API: To interact with the system to read, write, create, update and delete files 
**List 3 things that are part of the Jest API**
- Global Methods
- CLI commands
- JEST object
**List 3 things that are part of the Spotify API**
- Playback control
- Authorization
- Braodcasting
**Explain why Jest is not a web-based API and why Spotify is a web-based API.**
Ans. Jest is a testing framework which can work locally without the need of the internet while Spotify on the other hand is web based API which relies on the internet to work and cannot work without the internet

**Comment by areebsattar** [comment]
![Image](https://github.com/areebsattar/My-Coursework-Planner/assets/128264939/e3e1bedf-1afe-494a-ab1f-1dbee43d6444)

**Comment by SpectacularLandscape** [comment]
Great work, @areebsattar!
